agent/taskresource/envFiles/envfile.go

```diff
@@ -283,30 +284,48 @@ func (envfile *EnvironmentFileResource) Create() error {
 		return err
 	}
 
+	var wg sync.WaitGroup
+	errorEvents := make(chan error, len(envfile.environmentFilesSource))
+
 	iamCredentials := executionCredentials.GetIAMRoleCredentials()
 	for _, envfileSource := range envfile.environmentFilesSource {
+		wg.Add(1)
 		// if we support types besides S3 ARN, we will need to add filtering before the below method is called
+		// call an additional go routine per env file
+		go envfile.downloadEnvfileFromS3(envfileSource.Value, iamCredentials, &wg, errorEvents)
+	}
+
+	wg.Wait()
+	close(errorEvents)
 
-		err := envfile.downloadEnvfileFromS3(envfileSource.Value, iamCredentials)
-		if err != nil {
-			err = errors.Wrapf(err, "unable to download envfile with ARN %v from s3", envfileSource.Value)
-			envfile.setTerminalReason(err.Error())
-			return err
+	if len(errorEvents) > 0 {
+		var terminalReasons []string
+		for err := range errorEvents {
+			terminalReasons = append(terminalReasons, err.Error())
 		}
+
+		errorString := strings.Join(terminalReasons, ";")
+		envfile.setTerminalReason(errorString)
+		return errors.New(errorString)
 	}
 
 	return nil
 }
 
-func (envfile *EnvironmentFileResource) downloadEnvfileFromS3(envFilePath string, iamCredentials credentials.IAMRoleCredentials) error {
+func (envfile *EnvironmentFileResource) downloadEnvfileFromS3(envFilePath string, iamCredentials credentials.IAMRoleCredentials,
```

Comment:
in error cases, should we log the task id/arn or does it get logged in the method that calls this?

Reply:
the full s3 arn should get logged. should we also log the task arn?

Reply:
s3 arn doesn't provide any identifying info about for which task the env file download was attempted so yeah i think we should add it.